src/Array.hpp

```diff
@@ -385,91 +379,61 @@ class Array : public ArrayView< T,
   }
 
   /**
-   * @brief Append a value to the end of the array.
-   * @param value the value to append via a copy.
+   * @brief Construct a value in place at the end of the array.
+   * @tparam ARGS A variadic pack of the types to construct the new value from.
+   * @param args A variadic pack of values to construct the new value from.
    * @note This method is only available on 1D arrays.
    * @return void.
    */
-  template< int N=NDIM >
-  std::enable_if_t< N == 1 >
-  push_back( T const & value )
+  template< typename ... ARGS, int _NDIM=NDIM >
+  std::enable_if_t< _NDIM == 1 >
+  emplace_back( ARGS && ... args )
```

Comment:
`emplace_back` is a drop in replacement for `push_back` and it's supported by `std::vector` as well. I would prefer not to also have `push_back` because it would add two methods `push_back( T const &)` and `push_back( T && )` that add no new functionality.